auto-generated/vector-crypto/intrinsic_funcs/00_zvbb_-_vector_bit-manipulation_used_in_cryptography.md

```diff
@@ -0,0 +1,581 @@
+vuint16mf4_t __riscv_vror_vx_u16mf4_m (vbool64_t mask, vuint16mf4_t vs2, uint16_t rs1, size_t vl);
+vuint16mf2_t __riscv_vror_vv_u16mf2_m (vbool32_t mask, vuint16mf2_t vs2, vuint16mf2_t vs1, size_t vl);
+vuint16mf2_t __riscv_vror_vx_u16mf2_m (vbool32_t mask, vuint16mf2_t vs2, uint16_t rs1, size_t vl);
+vuint16m1_t __riscv_vror_vv_u16m1_m (vbool16_t mask, vuint16m1_t vs2, vuint16m1_t vs1, size_t vl);
+vuint16m1_t __riscv_vror_vx_u16m1_m (vbool16_t mask, vuint16m1_t vs2, uint16_t rs1, size_t vl);
+vuint16m2_t __riscv_vror_vv_u16m2_m (vbool8_t mask, vuint16m2_t vs2, vuint16m2_t vs1, size_t vl);
+vuint16m2_t __riscv_vror_vx_u16m2_m (vbool8_t mask, vuint16m2_t vs2, uint16_t rs1, size_t vl);
+vuint16m4_t __riscv_vror_vv_u16m4_m (vbool4_t mask, vuint16m4_t vs2, vuint16m4_t vs1, size_t vl);
+vuint16m4_t __riscv_vror_vx_u16m4_m (vbool4_t mask, vuint16m4_t vs2, uint16_t rs1, size_t vl);
+vuint16m8_t __riscv_vror_vv_u16m8_m (vbool2_t mask, vuint16m8_t vs2, vuint16m8_t vs1, size_t vl);
+vuint16m8_t __riscv_vror_vx_u16m8_m (vbool2_t mask, vuint16m8_t vs2, uint16_t rs1, size_t vl);
+vuint32mf2_t __riscv_vror_vv_u32mf2_m (vbool64_t mask, vuint32mf2_t vs2, vuint32mf2_t vs1, size_t vl);
+vuint32mf2_t __riscv_vror_vx_u32mf2_m (vbool64_t mask, vuint32mf2_t vs2, uint32_t rs1, size_t vl);
+vuint32m1_t __riscv_vror_vv_u32m1_m (vbool32_t mask, vuint32m1_t vs2, vuint32m1_t vs1, size_t vl);
+vuint32m1_t __riscv_vror_vx_u32m1_m (vbool32_t mask, vuint32m1_t vs2, uint32_t rs1, size_t vl);
+vuint32m2_t __riscv_vror_vv_u32m2_m (vbool16_t mask, vuint32m2_t vs2, vuint32m2_t vs1, size_t vl);
+vuint32m2_t __riscv_vror_vx_u32m2_m (vbool16_t mask, vuint32m2_t vs2, uint32_t rs1, size_t vl);
+vuint32m4_t __riscv_vror_vv_u32m4_m (vbool8_t mask, vuint32m4_t vs2, vuint32m4_t vs1, size_t vl);
+vuint32m4_t __riscv_vror_vx_u32m4_m (vbool8_t mask, vuint32m4_t vs2, uint32_t rs1, size_t vl);
+vuint32m8_t __riscv_vror_vv_u32m8_m (vbool4_t mask, vuint32m8_t vs2, vuint32m8_t vs1, size_t vl);
+vuint32m8_t __riscv_vror_vx_u32m8_m (vbool4_t mask, vuint32m8_t vs2, uint32_t rs1, size_t vl);
+vuint64m1_t __riscv_vror_vv_u64m1_m (vbool64_t mask, vuint64m1_t vs2, vuint64m1_t vs1, size_t vl);
+vuint64m1_t __riscv_vror_vx_u64m1_m (vbool64_t mask, vuint64m1_t vs2, uint64_t rs1, size_t vl);
+vuint64m2_t __riscv_vror_vv_u64m2_m (vbool32_t mask, vuint64m2_t vs2, vuint64m2_t vs1, size_t vl);
+vuint64m2_t __riscv_vror_vx_u64m2_m (vbool32_t mask, vuint64m2_t vs2, uint64_t rs1, size_t vl);
+vuint64m4_t __riscv_vror_vv_u64m4_m (vbool16_t mask, vuint64m4_t vs2, vuint64m4_t vs1, size_t vl);
+vuint64m4_t __riscv_vror_vx_u64m4_m (vbool16_t mask, vuint64m4_t vs2, uint64_t rs1, size_t vl);
+vuint64m8_t __riscv_vror_vv_u64m8_m (vbool8_t mask, vuint64m8_t vs2, vuint64m8_t vs1, size_t vl);
+vuint64m8_t __riscv_vror_vx_u64m8_m (vbool8_t mask, vuint64m8_t vs2, uint64_t rs1, size_t vl);
+```
+
+### [Vector Bit-manipulation used in Cryptography - Shift]():
```

Comment:
We have signed and unsigned versions of `vsll`, but this proposal only adds unsigned versions of `vwsll`. I'm not proposing a change, just mentioning this asymmetry.

Reply:
Yes I am also aware of the asymmetry. The vector crypto instructions generally operates under an unsigned data type and all other intrinsics does not have variants for signed data type. So I think it is more reasonable to not a signed data type variant here.

Reply:
`vwsll` will zero-extend `vs2` to 2x SEW, so I think no signed version might be less confused.